### PR TITLE
vkd3d: 1.10 -> 1.12

### PR DESCRIPTION
**Pull request by AndersonTorres**
## Description of changes

Closes https://github.com/NixOS/nixpkgs/pull/293902

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by SuperSamus** [comment]
Small note: vkd3d-proton 2.12 is likely to release soon, given that the changelog was already committed.

**Comment by AndersonTorres** [comment]
I will stack it when it arrives, then.

File: pkgs/by-name/vk/vkd3d-proton/package.nix [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+  nativeBuildInputs = [
+    glslang
+    meson
+    ninja
+    wine
+  ];
+
+  strictDeps = true;
+
+  prePatch = let
```
Comment (SuperSandro2000) [inline pkgs/by-name/vk/vkd3d-proton/package.nix:33]:
```suggestion
  postPatch = let
```
We don't want to break patches

File: pkgs/by-name/vk/vkd3d-proton/package.nix [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "vkd3d-proton";
+  version = "2.11.1-unstable-2024-03-07";
+
+  src = fetchFromGitHub {
+    owner = "HansKristian-Work";
+    repo = "vkd3d-proton";
+    rev = "ffb04fb7091f932ca95da236d63ab9a4064d6443";
+    fetchSubmodules = true;
```
Comment (SuperSandro2000) [inline pkgs/by-name/vk/vkd3d-proton/package.nix:18]: Is there a point in devendoring things?

Comment (AndersonTorres) [reply pkgs/by-name/vk/vkd3d-proton/package.nix:19]: Well, it can possibly work. I will try.

Comment (AndersonTorres) [reply pkgs/by-name/vk/vkd3d-proton/package.nix]: Splitted at https://github.com/NixOS/nixpkgs/pull/319877

File: pkgs/by-name/vk/vkd3d/package.nix [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+  };
+
+  outputs = [ "out" "dev" "lib" ];
+
+  nativeBuildInputs = [
+    autoreconfHook
+    bison
+    flex
+    pkg-config
+    wine
```
Comment (SuperSandro2000) [inline pkgs/by-name/vk/vkd3d/package.nix:34]: wine as a nativeBuildInput? I am a bit surprised but with strictDeps it is probably fine.

**Review by SuperSandro2000 — APPROVED** [review]
LGTM

File: pkgs/by-name/vk/vkd3d/package.nix (hunk 3 above)
Comment (AndersonTorres) [reply pkgs/by-name/vk/vkd3d/package.nix:34]: A typical SDL situation...

Comment (SuperSandro2000) [reply pkgs/by-name/vk/vkd3d/package.nix:34]: I am hearing for the first time that this is SDL related. How does it come?

Comment (AndersonTorres) [reply pkgs/by-name/vk/vkd3d/package.nix:34]: I said in the sense SDL requires both nativeBuildInputs and buildInputs, because each has a distinct role.

**Pull request merged**